File: backend/src/main/java/dev/agiro/fanel/calendar/api/CalendarApi.java
package dev.agiro.fanel.calendar.api;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.List;
import java.util.UUID;

public interface CalendarApi {
    /** Occurrences (recurring events expanded) within [from, to]; defaults to roughly the last month through the next year when null. */
    List<CalendarEventDto> list(UUID householdId, LocalDate from, LocalDate to);
    /** Raw series, one row per event (no recurrence expansion); used for export. */
    List<CalendarEventDto> listAll(UUID householdId);
    CalendarEventDto create(UUID householdId, String title, LocalDate date, LocalTime time, UUID addedBy,
                            List<UUID> assigneeIds, RecurrenceFrequency recurrenceFreq, Integer recurrenceInterval,
                            LocalDate recurrenceUntil);
    CalendarEventDto update(UUID householdId, UUID eventId, String title, LocalDate date, LocalTime time,
                            List<UUID> assigneeIds, RecurrenceFrequency recurrenceFreq, Integer recurrenceInterval,
                            LocalDate recurrenceUntil);
    void delete(UUID householdId, UUID eventId);
}

File: android/app/src/main/java/dev/agiro/fanel/android/data/remote/CalendarApi.kt
package dev.agiro.fanel.android.data.remote

import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.PATCH
import retrofit2.http.POST
import retrofit2.http.Path
import retrofit2.http.Query

interface CalendarApi {
    @GET("/api/v1/households/{householdId}/calendar")
    suspend fun list(
        @Path("householdId") householdId: String,
        @Query("from") from: String,
        @Query("to") to: String
    ): List<CalendarEventDto>

    @POST("/api/v1/households/{householdId}/calendar")
    suspend fun create(
        @Path("householdId") householdId: String,
        @Body request: CreateEventRequest
    ): CalendarEventDto

    @PATCH("/api/v1/households/{householdId}/calendar/{eventId}")
    suspend fun update(
        @Path("householdId") householdId: String,
        @Path("eventId") eventId: String,
        @Body request: UpdateEventRequest
    ): CalendarEventDto

    @DELETE("/api/v1/households/{householdId}/calendar/{eventId}")
    suspend fun delete(
        @Path("householdId") householdId: String,
        @Path("eventId") eventId: String
    )
}
